app/src/main/java/com/example/tungcon/demo/movie/MovieActivity.java

```diff
@@ -0,0 +1,61 @@
+package com.example.tungcon.demo.movie;
+
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+
+
+import com.example.tungcon.demo.R;
+import com.example.tungcon.demo.data.model.Movie;
+import com.example.tungcon.demo.data.repository.MovieRepository;
+import com.example.tungcon.demo.movie.IamgeMovie.ItemMovieActivity;
+
+import java.util.ArrayList;
+
+
+public class MovieActivity extends AppCompatActivity implements MovieContact.View,
+        MovieAdapter.OnMovieItemClickListener {
+    private MovieAdapter mMovieAdapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_movie);
+        loadData();
+        initView();
+    }
+
+    private void initView() {
+        initRecycler();
+    }
+
+    private void initRecycler() {
+        RecyclerView mRecycleMovie = findViewById(R.id.recycler_movie);
+        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this);
+        mRecycleMovie.setLayoutManager(layoutManager);
+        mMovieAdapter = new MovieAdapter(this);
+        mRecycleMovie.setAdapter(mMovieAdapter);
+    }
+
+    private void loadData() {
+        MovieContact.Presenter presenter = new MoviePresenterImp(
+                this,
+                MovieRepository.getInstance());
+        presenter.getMovies();
+    }
+
+
+    @Override
+    public void showMovies(ArrayList<Movie> data) {
+        mMovieAdapter.setMovies(data);
```

Comment:
e nên viết 1 method addData cho adapter và gọi notifiy ở trong đó như vậy gọn code hơn.
```suggestion
        mMovieAdapter.addData(data);
```